api/envoy/extensions/original_ip_detection/xff/v3/xff.proto

```diff
@@ -0,0 +1,25 @@
+syntax = "proto3";
+
+package envoy.extensions.original_ip_detection.xff.v3;
+
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.original_ip_detection.xff.v3";
+option java_outer_classname = "XffProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: XFF original IP detection extension]
+
+// This extension allows for the original downstream remote IP to be detected
+// by reading the :ref:`config_http_conn_man_headers_x-forwarded-for` header.
+//
+// [#extension: envoy.original_ip_detection.xff]
+message XffConfig {
+  // The number of additional ingress proxy hops from the right side of the
+  // :ref:`config_http_conn_man_headers_x-forwarded-for` HTTP header to trust when
+  // determining the origin client's IP address. The default is zero if this option
+  // is not specified. See the documentation for
+  // :ref:`config_http_conn_man_headers_x-forwarded-for` for more information.
+  uint32 xff_num_trusted_hops = 1;
```

Comment:
Hmmm, I'm inclined to think we should pass hcm XFF to all extensions for them to base decisions on.  That said ideally I think it'd be trusted hops, not xff_trusted_hops, then you could cross-check that value in your trusted-header extension (either ignore that value if trusted_hops == 0, or config valiate that if you're configuring that extension that trusted hops >=0)

Mainly I want to avoid the issue where we have xff_trusted hops here and in the hcm so if we're adding here I think we need to plain to deprecate the HCM location.  WDYT?

Reply:
Er if an extension really needs to know about XFF and stuff they can just call `Utility::getLastAddressFromXFF` and be done (and have a knob for it), no? I am wondering if we are giving XFF too much preponderance.

For our use case, we certainly don't care about XFF... 

But it sounds like for your use case passing XFF around might be important? So happy to follow your lead here.

Reply:
In a perfect world I think we would deprecate the HCM field for this, and force it to be set here. I think in the case where it's set in the HCM and no extensions are set, we could instantiate an XFF config with this field set? wdyt?

Reply:
I'm fine deprecating the HCM field, and having a transition over (validate they're not both set and unequal).  I just really don't want to allow both long term.

Reply:
@mattklein123: yes, that's what we do now pretty much:

```
  if (!ip_detection_extensions.empty()) {
     ...
  } else {
    original_ip_detection_extensions_.push_back(
        std::make_shared<OriginalIPDetection::Xff::XffIPDetection>(xff_num_trusted_hops_));
  }
```

@alyssawilk: yep I agree, I'll mark `xff_num_trusted_hops` as deprecated and point out that in the future this needs to be setup via `original_ip_detection_extensions`.